examples/hallowelt/go_app/README.md

```diff
@@ -0,0 +1,253 @@
+	"os"
+
+	fpc "github.com/hyperledger-labs/fabric-private-chaincode/client_sdk/go/pkg/gateway"
+	"github.com/hyperledger-labs/fabric-private-chaincode/integration/client_sdk/go/utils"
+	"github.com/hyperledger/fabric-sdk-go/pkg/gateway"
+	"github.com/hyperledger/fabric/common/flogging"
+)
+```
+
+* We will also define several variables. `network` represents the Fabric network and `contract` our fabric private chaincode developed previously. We also are using a variable to be able to capture the errors that the different functions might return.
+
+```go
+var (
+	network  *gateway.Network
+	contract fpc.Contract
+	err      error
+	logger   = flogging.MustGetLogger("helloworld")
+)
+```
+* We will then add the main function which will execute our code. The environent variable `CC_ID` represents the FPC chaincode already installed on the channel and will be set from the bash script we will create to run evenrything. The Fabric network we are setting up has only one channel called `mychannel` which we will connect to, using the `SetupNetwork` go method provided in the `utils` library we imported before. Then we will access the FPC chaincode and call the `SubmitTransaction` go method that uses the first parameter as the chaincode method to invoke, the second as the name of the asset and the third as the value. In this case we will be storing 100 as the value of `asset1`  invoking the `storeAsset` method.
+
+```go
+func main() {
+	ccID := os.Getenv("CC_ID")
+	logger.Infof("Environment variable CC_ID: %v", ccID)
+	// get network
+	network, err = utils.SetupNetwork("mychannel")
+
+	// Get FPC Contract
+	contract := fpc.GetContract(network, ccID)
+
+	// Invoke FPC Chaincode storeAsset
+	logger.Infof("--> Invoke FPC chaincode storeAsset: ")
+	result, err := contract.SubmitTransaction("storeAsset", "asset1", "100")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+}
+)
+```
+* To be able to query the ledger we have to call the same method changing the first parameter to `retrieveAsset` and since we are querying there is no need for the third parameter.
+```go
+// Invoke FPC Chaincode retrieveAsset
+	logger.Infof("--> Invoke FPC chaincode retrieveAsset: ")
+	result, err = contract.SubmitTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+* Both previous transactions have been registered on the ledger, to be able to query without this occurring we will use the `EvaluateTransaction` go method instead.
+```go
+	// Evaluate FPC Chaincode retrieveAsset
+	logger.Infof("--> Evaluate FPC chaincode retrieveAsset: ")
+	result, err = contract.EvaluateTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Evaluate transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+
+The whole text of the program would be
```

Comment:
While reading I had the feeling that we could provide the entire program code in a single snippet rather than splitting it as we did in the hello world tutorial.